### PR TITLE
feat(added-secure-prop-to-proxy-config) - added new parameter "secure…

### PR DESCRIPTION
**Pull request by goququ**
…" to be able to handle

handle "UNABLE_TO_VERIFY_LEAF_SIGNATURE" error

## 🚨 Proposed changes

> Please review the [guidelines for contributing](../../CONTRIBUTING.md) to this repository.

Hi, team! I am trying to use this plugin on my project and I have faced with some trouble to make it work with our instance of gitlab. So in general I want to be able to configure `secure` parameter according [this comment](https://github.com/chimurai/http-proxy-middleware/issues/233#issuecomment-355830567)

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

File: packages/gitlab-backend/src/service/router.ts [hunk 1]
```diff
@@ -21,12 +21,17 @@ function getBasePath(config: Config) {
...
 }
 
 export async function createRouter(
     options: RouterOptions
 ): Promise<express.Router> {
-    const { logger, config } = options;
+    const {
+        logger,
+        config,
+        secure = process.env.NODE_TLS_REJECT_UNAUTHORIZED !== '0',
```
Comment (antoniomuso) [inline packages/gitlab-backend/src/service/router.ts:33]: This parameter should be added to the configuration. In this way you can get it in this way: 
```ts
config.getOptionalBoolean('gitlab.proxySecure');
```

**Review by antoniomuso — CHANGES_REQUESTED** [review]

**Review by antoniomuso — CHANGES_REQUESTED** [review]
To use `config.getOptionalBoolean('gitlab.proxySecure')` you have to add the parameter in the [config.d.ts](https://github.com/immobiliare/backstage-plugin-gitlab/blob/main/packages/gitlab-backend/config.d.ts)

**Comment by goququ** [comment]
@antoniomuso hi, everything ok? Or I can do something else?

**Comment by antoniomuso** [comment]
LGTM! But the type checking fails...

**Comment by goququ** [comment]
> LGTM! But the type checking fails...

I haven‘t touched any react code. Do you think problem in my changes?

**Review by antoniomuso — APPROVED** [review]

**Comment by antoniomuso** [comment]
hi @goququ, this MR in up with [v5.2.0](https://github.com/immobiliare/backstage-plugin-gitlab/releases/tag/v5.2.0). Thanks for your contribution :smile:

**Pull request merged**